### PR TITLE
chore: remove bundlesize, use custom script

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

Follow-up to #3145.

Removes `bundlesize`, which seems unmaintained. Replaces it with a custom script that does the same thing.

No more need to wait for the slow `iltorb` dependency to compile (which we don't even need, since we don't check the Brotli size).

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

### CODE REVIEW

**Comment by nolanlawson** [comment]
/nucleus ignore --reason "downstreams failing for unrelated reason"

File: scripts/bundlesize/bundlesize.mjs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const jsonFile = path.join(__dirname, 'bundlesize.config.json');
+const { files } = JSON.parse(await fs.readFile(jsonFile, 'utf-8'));
+
+let failed = false;
+
+for (const { path: filePath, maxSize } of files) {
+    const absoluteFile = path.join(__dirname, '../..', filePath);
+    const source = await fs.readFile(absoluteFile, 'utf-8');
+    const { code } = await minify(source, { compress: true, mangle: true });
```
Comment (ravijayaramappa) [inline scripts/bundlesize/bundlesize.mjs:27]: Why is it necessary to re-minify the `.min.js` files? (_They are minified using terser originally_)

**Review by ravijayaramappa — COMMENTED** [review]
LGTM, I had one question though

File: scripts/bundlesize/bundlesize.mjs (hunk 1 above)
Comment (nolanlawson) [reply scripts/bundlesize/bundlesize.mjs]: You are correct, my mistake. Fixed.

**Review by ravijayaramappa — APPROVED** [review]

**Pull request merged**